oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextDocumentMaker.java

```diff
@@ -79,6 +90,9 @@ public FulltextDocumentMaker(@Nullable FulltextBinaryTextExtractor textExtractor
         this.path = checkNotNull(path);
         this.logWarnStringSizeThreshold = Integer.getInteger(WARN_LOG_STRING_SIZE_THRESHOLD_KEY,
                 DEFAULT_WARN_LOG_STRING_SIZE_THRESHOLD_VALUE);
+        this.throttleWarnLogs = Boolean.getBoolean(THROTTLE_WARN_LOGS_KEY);
+        this.throttleWarnLogThreshold = Integer.getInteger(THROTTLE_WARN_LOGS_THRESHOLD_KEY,
+                DEFAULT_THROTTLE_WARN_LOGS_THRESHOLD_VALUE);
```

Comment:
If I understood correctly, a new instance of this class is created for every node that is indexed. Adding two more instance fields here will increase memory and CPU usage during indexing, because this constructor is on the critical path of indexing. Aren't these values constant? Can we make them static variables?

Reply:
good point - I made them static variables.